# Next Generation Sequencing Workflow

### Requirement

* Python 2.7

### Description

| Code | Description | 
|---|---|
| preprocess.py | The code combined (1) dividing origin alignmant file from bowtie into several subfiles (it depends on the size of memory) and (2) combining the above extractLargeData.pl, getLargeData.pl and cbnCT.pl with the Linux commands. So this code must be executed on the Linux environment. |
| extract_data.py | The code was optional for executing and trying to extract the transcript name, gene name from the transcript label so as to simplify the output generated by analysis.r and to read the data more easily. |
| filter_database.py | Due to the limitation of the memory, the annotated file (.GTF) from GENCODE would be impossible to load the entire file. The code was designed to extract the potential data from the annotated database on the basis of the gene name in getTranscript.r. The process was first to get total gene name from the result generated by getTranscript.r and then executed the pattern matching in annotated file to extract potential entries to a new database. |



